### PR TITLE
Fix Mocks for Windows

### PR DESCRIPTION
**Pull request by osterman**
## what
- Run commands depending on the flavor of OS
- Add e2e smoke tests to verify commands work

## why
- Windows behaves differently from Linux/macOS
- #875 introduced windows testing
- Catch certain cases when we have no unit tests for CLI behaviors (tests for condition @mcalhoun encountered with an infinite loop in `atmos version`)

## Summary by CodeRabbit

- **New Features**
	- Introduced a comprehensive testing framework for the CLI commands.
	- Added a series of test cases for various functionalities of the `atmos` CLI, including error handling and command outputs.

- **Bug Fixes**
	- Improved conditional execution of tests based on operating system.
	- Enhanced clarity in test execution step names.

- **Chores**
	- Updated testing workflow to differentiate commands for Windows, Linux, and macOS.
	- Added steps to set the build artifacts directory in the PATH for both Linux/macOS and Windows targets.
	- Refined the process for downloading build artifacts with OS-specific naming.
	- Added GNU tar to PATH for improved build efficiency on Windows targets.
	- Restructured jobs to utilize a matrix strategy for better organization.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -279,28 +283,45 @@ jobs:
...
+        if: matrix.flavor.target == 'linux' || matrix.flavor.target == 'macos'
         run: |
-          cd examples/${{ matrix.demo-folder }}
           atmos test
 
+      - name: Run tests for ${{ matrix.demo-folder }} with ${{ matrix.flavor.target }}
+        working-directory: examples/${{ matrix.demo-folder }}
+        if: matrix.flavor.target == 'windows'
+        run: |
+          atmos.exe test
```
Comment (osterman) [inline .github/workflows/test.yml:323]: @coderabbitai when this workflow runs, atmos.exe is not found despite path being set. Any ideas?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -279,28 +283,58 @@
...
+        run: |
+          echo "${{ github.workspace }}" >> $GITHUB_PATH
+          chmod +x "${{ github.workspace }}/atmos
 
-      - name: Check out code into the Go module directory
-        uses: actions/checkout@v4
+      - name: Add build artifacts directory to PATH for ${{ matrix.flavor.target }}
+        if: matrix.flavor.target == 'windows'
+        run: |
+          echo "${{ github.workspace }}" >> $Env:GITHUB_PATH
```
[the remark this answers is not in the document]
Comment (osterman) [reply .github/workflows/test.yml:308]: I don't like switching from GITHUB_PATH to GITHUB_ENV, when all the docs say to use GITHUB_PATH

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -40,6 +40,10 @@ jobs:
       - name: Build
         run: echo "Building on ${{ matrix.os }}"
 
+      - name: Add GNU tar to PATH (significantly faster than windows tar)
+        if: matrix.target == 'windows'
+        run: echo "C:\Program Files\Git\usr\bin" >> $Env:GITHUB_PATH
+
```
[the remark this answers is not in the document]
Comment (osterman) [reply .github/workflows/test.yml:46]: All the examples I see use GITHUB_PATH, not GITHUB_ENV

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -296,11 +297,18 @@ jobs:
...
         run: |
-          cd examples/${{ matrix.demo-folder }}
           atmos test
 
+      - name: Run tests for ${{ matrix.demo-folder }} with ${{ matrix.flavor.target }}
+        working-directory: examples/${{ matrix.demo-folder }}
+        if: matrix.flavor.target == 'windows'
+        run: |
+          atmos.exe test
+
```
[the remark this answers is not in the document]
Comment (osterman) [reply .github/workflows/test.yml]: Impractical to maintain one version for both windows and linux/mac.

**Review by mcalhoun — APPROVED** [review]
LGTM

**Pull request merged**